Updated SSO intro (#871)
wandb/docs: docs/guides/hosting/iam/sso.md

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> W&B Server’s support for OpenID Connect (OIDC) compatible Identity Providers allows for streamlined and secure management of user identities and group memberships through external identity providers like Okta, Keycloak, Auth0, Google, or Microsoft’s Entra. This integration centralizes authentication, simplifying user management and ensuring adherence to security best practices such as single sign-on (SSO) and multi-factor authentication (MFA).

Reviewer: ```suggestion
W&B Server's support for OpenID Connect (OIDC) compatible identity providers allows for management of user identities and group memberships through external identity providers like Okta, Keycloak, Auth0, Google, and Entra.
```

> ## Open ID Connect

Reviewer: ```suggestion
## OpenID Connect (OIDC)
```

W&B Server supports two OpenID Connect (OIDC) authentication flows for integrating with external Identity Providers (IdPs): 
- Implicit Flow with Form Post and 

> - Authorization Code Flow with Proof Key for Code Exchange (PKCE). 

Reviewer: ```suggestion
W&B Server supports the following OIDC authentication flows for integrating with external Identity Providers (IdPs).
1. Implicit Flow with Form Post 
2. Authorization Code Flow with Proof Key for Code Exchange (PKCE)
```

> These flows are used to authenticate users and provide W&B Server with the necessary identity information (in the form of ID tokens) to manage access control.

Reviewer: ```suggestion
These flows authenticate users and provide W&B Server with the necessary identity information (in the form of ID tokens) to manage access control.
```

> The ID token is a JWT that contains the user’s identity information, including attributes such as name, username, email, and optionally group memberships. This is the primary token W&B Server uses to authenticate the user and map them to appropriate roles or groups in the system.

Reviewer: ```suggestion
The ID token is a JWT that contains the user's identity information, such as their name, username, email, and group memberships. W&B Server uses this token to authenticate the user and map them to appropriate roles or groups in the system.
```

> In the context of W&B Server, the access token is not required. Access tokens are typically used to authorize requests to APIs on behalf of the user, but since W&B Server’s primary concern is user authentication and identity, only the ID token is needed.

Reviewer: ```suggestion
In the context of W&B Server, the access token is not required. Access tokens are typically used to authorize requests to APIs on behalf of the user, but since W&B Server’s primary concern is user authentication and identity, it only requires the ID token.
```

> In addition to basic [environment variables](../env-vars.md), you can use environment variables to [configure IAM options](advanced_env_vars.md) for your [Dedicated Cloud](../hosting-options/dedicated_cloud.md) or [Self-managed](../hosting-options/self-managed.md) instance.

Reviewer: ```suggestion
You can use environment variables to [configure IAM options](advanced_env_vars.md) for your [Dedicated Cloud](../hosting-options/dedicated_cloud.md) or [Self-managed](../hosting-options/self-managed.md) instance.
```
